### PR TITLE
Review blog

### LINKED ISSUE #16: [API] POST admin/blog/review/:id
@@Admin

Ensure user is admin based on #15 

send review of blog based on:

- `pending`: no review done
- `approved`: blog is published for everyone
- `update`: improve blog based on `review_message`

### PR DESCRIPTION
**Pull request by RiyaRaj28**
## @issue16 : Review Blog 

Resolves [Issue 16](https://github.com/technobyte-nitkkr/Blogs/issues/16#issue-2321551649). 

Added logic for blog review by admin. 

* Sending a review of the blog based on pending, approved or updated.

### CODE REVIEW

File: server/src/controllers/blog-review.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  const { reviewStatus, reviewMessage } = req.body;
+
+
+  try {
+  const blog = await Blog.findById(blogId);
+  if (!blog) {
+    return res.status(404).json({ message: 'Blog not found' });
+  }
+
+  const user = req.user;
```
Comment (Himasnhu-AT) [inline server/src/controllers/blog-review.js]: implemented in a later stage, not required right now. Will cause the api to fail

File: server/src/controllers/blog-review.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  }
+
+  if (user.role != 'admin') {
+    return res.status(403).json({ message: 'Only admins can review blogs' });
+  }
+
+  switch (reviewStatus) {
+    case 'pending':
+      blog.review = 'pending';
+      break;
```
Comment (Himasnhu-AT) [inline server/src/controllers/blog-review.js]: will cause server to crash

File: server/src/controllers/blog-review.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    case 'approved':
+      blog.review = 'approved';
+      blog.isPublished = true;
+      res.json({ message: 'Blog approved!' });
+      break;
+    case 'update':
+      blog.review = 'update';
+      blog.reviewMessage = reviewMessage;
+      res.json({ message: `Blog requires updates. Please review the comments: ${reviewMessage}` });
+      break;
```
Comment (Himasnhu-AT) [inline server/src/controllers/blog-review.js]: crashing server

File: server/src/controllers/blog-review.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    default:
+      return res.status(400).json({ message: 'Invalid review status' });
+  }
+
+  await blog.save();
+
+  res.json({ message: 'Blog review updated successfully' });
```
Comment (Himasnhu-AT) [inline server/src/controllers/blog-review.js]: won't even reach this line. as response is already allocated above

File: server/src/model/blog.js [hunk 5]
```diff
@@ -1,5 +1,5 @@
 import mongoose from 'mongoose'
-import Tag from './tag.js'
+import Tag from './tags.js'
```
Comment (Himasnhu-AT) [inline server/src/model/blog.js:2]: follow the proper linting method.  enter after function change for making reading easier

**Review by Himasnhu-AT — CHANGES_REQUESTED** [review]
@RiyaRaj28  Complete the comments below to make it working. Also check proper functionality by imitating server operation by hardcoding things

Resolving [Issue 16](https://github.com/technobyte-nitkkr/Blogs/issues/16#issue-2321551649).

**Comment by Debatreya** [comment]
@Himasnhu-AT 
Folder structure as discussed by.us.
Usme hum src/routes mein hi controller likh rhe the
 and src/routes/routes.js mein route declare kr rhe the. 

Ig we should get rid of the src/controller.

Bta dena.
Mere issue wala kaam ho jaay, 
Mein structure theek kr dunga.

**Comment by RiyaRaj28** [comment]
> @RiyaRaj28 Complete the comments below to make it working. Also check proper functionality by imitating server operation by hardcoding things
> 
> Resolving [Issue 16](https://github.com/technobyte-nitkkr/Blogs/issues/16#issue-2321551649).



> @RiyaRaj28 Complete the comments below to make it working. Also check proper functionality by imitating server operation by hardcoding things
> 
> Resolving [Issue 16](https://github.com/technobyte-nitkkr/Blogs/issues/16#issue-2321551649).

Okay.

**Review by Himasnhu-AT — CHANGES_REQUESTED** [review]
Great Work!

### UPDATE: 
When setting `reviewStatus` to update, make sure message is provided and is not undefined

API CALL:

- Request: 
```bash
ssh:OpenEdu_AI_test Blogs % curl -X POST http://localhost:3000/admin/blog/review/668a67885a968b26ecb519a5 -H "Content-Type: application/json" -d '{"reviewStatus": "update"}' 
```
- Response:
```json
{
  "message": "Blog requires updates. Please review the comments: undefined",
  "updatedBlog": {
    "_id": "668a67885a968b26ecb519a5",
    "title": "My New Blog",
    "content": "This is my new blog content",
    "image": "https://example.com/image.jpg",
    "review": "update",
    "tags": [],
    "comments": [],
    "__v": 0
  }
}
```

**Comment by Himasnhu-AT** [comment]
Also, convert `admin/blog` API to `seed`

@Debatreya, Let's keep this function to seed the data. Let other know to enter the seed for testing purposes.

---

> @Himasnhu-AT Folder structure as discussed by.us. Usme hum src/routes mein hi controller likh rhe the and src/routes/routes.js mein route declare kr rhe the.
> 
> Ig we should get rid of the src/controller.
> 
> Bta dena. Mere issue wala kaam ho jaay, Mein structure theek kr dunga.

Chl thik hai, krdio jo structure aacha lge

**Comment by Debatreya** [comment]
Raat ko call krta hu

**Review by Himasnhu-AT — APPROVED** [review]

**Pull request merged**